File: Java/Mit Polymorphismus/Katze.java
public class Katze extends Haustier{

    public Katze(String n){
        super(n);
    }

    public void fressen(){
        System.out.println("Maus fressen");
    }

    public void freuen(){
        System.out.println("schnurren");
    }

    public void lautGeben(){
        System.out.println("miauen");
    }
}

File: TypeScript/Mit Polymorphismus/Katze.ts
import {Haustier} from "./Haustier";

class Katze extends Haustier{
    constructor(n : string){
        super(n);

    }

    fressen(){
        console.log("Maus fressen");
    }

    freuen(){
        console.log("schnurren");
    }

    lautGeben(){
        console.log("miauen");
    }
}

export {Katze};

